javascript/GridField.js

```diff
@@ -146,6 +152,37 @@
 			}
 		});
 
+		/**
+		 * Don't allow users to submit empty values in grid field auto complete inputs.
+		 */
+		$('.ss-gridfield .add-existing-autocompleter').entwine({
+			onbuttoncreate: function () {
+				var self = this;
+
+				this.toggleDisabled();
+
+				this.find('input[type="text"]').bind('keyup', function () {
+					self.toggleDisabled();
+				});
+			},
+			onunmatch: function () {
+				this.find('input[type="text"]').unbind('keyup');
+			},
+			toggleDisabled: function () {
+				var $button = this.find('button[type="submit"]'),
+					$input = this.find('input[type="text"]'),
+					inputValue = $input.val(),
+					buttonDisabled = $button.prop('disabled');
+
+				if ((inputValue === '' && buttonDisabled === false) ||
```

Comment:
It would probably work fine without any condition here. Does it save any work if this evaluates to false?